### PR TITLE
DIG-1511: Query beacon results should only return authorized results

### PR DESCRIPTION
**Pull request by daisieh**
I actually found a bigger, long-ago bug while investigating this: Long ago, I'd assumed that site admins should be able to see all objects in all cohorts; for example, in our usual tests, user2 would be able to see things from SYNTHETIC-1 and SYNTHETIC-2, even if it is only explicitly authorized for SYNTHETIC-2. This is now fixed.

Anyway: now if you try to replicate the conditions in https://candig.atlassian.net/browse/DIG-1510, you should get the correct results.

I also had to tweak tests to add the test cohorts to Opa, if Opa is available (i.e. CANDIG_URL is available). This is required for integration tests, which were overly reliant on the site admin user being able to see everything.

NB: if you want to run integration tests on your system and have them pass after this, you'll need to pull https://github.com/CanDIG/CanDIGv2/pull/478 and also make sure your version of Opa has pulled https://github.com/CanDIG/candig-opa/pull/49.

To sum up: make sure you've pulled opa to develop, then do `make clean-opa; make docker-volumes; make build-opa; make compose-opa`. Then rebuild htsget and then run integration tests.

### CODE REVIEW

**Review by OrdiNeu — APPROVED** [review]
LGTM :heavy_check_mark:
